crypto/fipsmodule/ec/asm/p256-armv8-asm.pl

```diff
@@ -1583,157 +1583,111 @@
+    eor     $Rf.16b, $Rf.16b, $Rf.16b
 
-    // $M0 := $ONE
-    mov     $M0.16b, $ONE.16b               ////    movdqa  $ONE, $M0
+.Lselect_w5_loop:
+    // Loop 16 times.
 
-    // $Val_in := $val
-    // $Ctr := 16; loop counter
-    mov     $Val_in, $val
-    mov     $Ctr, #16                       ////    mov \$16, %rax
+    // Increment index (loop counter); tested at the end of the loop
+    add $Idx_ctr, $Idx_ctr, #1
 
-.Lselect_w5_loop:
     // [$T0a-$T0f] := Load a (3*256-bit = 6*128-bit) table entry starting at $in_t
     //  and advance $in_t to point to the next entry
     ld1     {$T0a.2d, $T0b.2d, $T0c.2d, $T0d.2d}, [$in_t],#64
-    ld1     {$T0e.2d, $T0f.2d}, [$in_t],#32 ////    movdqa  16*0($in_t), $T0a
-                                            ////    movdqa  16*1($in_t), $T0b
-                                            ////    movdqa  16*2($in_t), $T0c
-                                            ////    movdqa  16*3($in_t), $T0d
-                                            ////    movdqa  16*4($in_t), $T0e
-                                            ////    movdqa  16*5($in_t), $T0f
-                                            ////    lea 16*6($in_t), $in_t
-
-    // $TMP0 = ($M0 == $INDEX)? All 1s : All 0s
-    cmeq    $TMP0.4s, $M0.4s, $INDEX.4s     ////    movdqa  $M0, $TMP0
-                                            ////    pcmpeqd $INDEX, $TMP0
-    // Increment $M0 lanes
-    add     $M0.4s, $M0.4s, $ONE.4s         ////    paddd   $ONE, $M0
-
-    // [$T0a-$T0f] := [$T0a-$T0f] AND $TMP0
-    // values read from the table will be 0'd if $M0 != $INDEX
-    // [$Ra-$Rf] := [$Ra-$Rf] OR [$T0a-$T0f]
-    // values in output registers will remain the same if $M0 != $INDEX
-    and     $T0a.16b, $T0a.16b, $TMP0.16b    ////    pand   $TMP0, $T0a
-    and     $T0b.16b, $T0b.16b, $TMP0.16b    ////    pand   $TMP0, $T0b
-    orr     $Ra.16b, $Ra.16b, $T0a.16b       ////    por    $T0a, $Ra
-    orr     $Rb.16b, $Rb.16b, $T0b.16b       ////    pand   $TMP0, $T0c
-                                             ////    por    $T0b, $Rb
-    and     $T0c.16b, $T0c.16b, $TMP0.16b    ////    pand   $TMP0, $T0d
-    and     $T0d.16b, $T0d.16b, $TMP0.16b    ////    por    $T0c, $Rc
-    orr     $Rc.16b, $Rc.16b, $T0c.16b       ////    pand   $TMP0, $T0e
-    orr     $Rd.16b, $Rd.16b, $T0d.16b       ////    por    $T0d, $Rd
-                                             ////    pand   $TMP0, $T0f
-    and     $T0e.16b, $T0e.16b, $TMP0.16b    ////    por    $T0e, $Re
-    and     $T0f.16b, $T0f.16b, $TMP0.16b    ////    por    $T0f, $Rf
-    orr     $Re.16b, $Re.16b, $T0e.16b
-    orr     $Rf.16b, $Rf.16b, $T0f.16b
-
-    // Decrement loop counter; loop back if not 0
-    subs    $Ctr, $Ctr, #1                  ////    dec %rax
-    bne     .Lselect_w5_loop                ////    jnz .Lselect_loop_sse_w5
+
+    // $Mask_64 := ($Idx_ctr == $index)? All 1s : All 0s
+    cmp     $Idx_ctr, $index
+    csetm   $Mask_64, eq
```

Comment:
`bit` instructions have no effect on the result `$Ra.16b` when `$Mask.16b` is zero.
We can stop executing the loop body as soon as we know `$Mask.16b` is zero, which is after the above `cmp $Idx_ctr, $index` instruction.
If `$Idx_ctr != $index` we could just bump the address `add $in_t, $in_t, #94` and go to next iteration.
That would remove both `ld1` instructions and all the `bit` instructions below.